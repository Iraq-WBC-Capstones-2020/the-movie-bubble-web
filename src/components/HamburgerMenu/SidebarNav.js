import React, { useRef } from 'react';
import { motion, useCycle } from 'framer-motion';
import useDimensions from './useDimensions';
import Toggler from './Toggler';
import NavList from './NavList';
const sidebar = {
  open: (height = 1000) => ({
    clipPath: `circle(${height * 2 + 200}px at 86.8% 40px)`,
    transition: {
      type: 'spring',
      stiffness: 30,
      restDelta: 2,
    },
  }),
  closed: {
    clipPath: 'circle(28px at 84.5% 40px)',
    transition: {
      type: 'spring',
      stiffness: 550,
      damping: 40,
    },
  },
};

export default function SidebarNav() {
  const [isOpen, toggleOpen] = useCycle(false, true);
  const containerRef = useRef(null);
  const { height } = useDimensions(containerRef);
  return (
    <>
      <motion.nav
        className="absolute	 top-0 right-0 w-64 bottom-0 overflow-hidden"
        initial={false}
        animate={isOpen ? 'open' : 'closed'}
        custom={height}
        ref={containerRef}
      >
        <motion.div
          style={{
            background: `linear-gradient(270deg, #FF7521 11.7%, #FFB421 90.81%)`,
          }}
          className="absolute	 top-0 right-0 w-64 bottom-0 "
          variants={sidebar}
        />
        <NavList />

        <Toggler toggle={() => toggleOpen()} />
      </motion.nav>
    </>
  );
}
