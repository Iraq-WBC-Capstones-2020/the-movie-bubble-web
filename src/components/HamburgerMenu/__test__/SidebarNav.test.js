import React from 'react';
import SidebarNav from '../SidebarNav';
import renderer from 'react-test-renderer';
test('Humburger menu snapshot testd succesfuly', () => {
  const tree = renderer.create(<SidebarNav />).toJSON();
  expect(tree).toMatchSnapshot();
});
