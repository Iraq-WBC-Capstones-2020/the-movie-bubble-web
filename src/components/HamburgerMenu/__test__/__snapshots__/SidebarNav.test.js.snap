// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Humburger menu snapshot testd succesfuly 1`] = `
<nav
  className="absolute	 top-0 right-0 w-64 bottom-0 overflow-hidden"
  style={Object {}}
>
  <div
    className="absolute	 top-0 right-0 w-64 bottom-0 bg-orange-400"
    style={
      Object {
        "clipPath": "circle(28px at 84.5% 40px)",
      }
    }
  />
  <ul
    className="p-2 absolute  w-full flex flex-col items-center h-full "
    style={
      Object {
        "display": "none",
        "opacity": 0,
        "transform": "translateY(50px) translateZ(0)",
      }
    }
  >
    <li
      className="bg-gray-900 w-32 text-center mt-48  my-5 cursor-pointer rounded"
      style={Object {}}
    >
      <button
        className="w-full py-2 focus:outline-none text-white"
      >
        Home
      </button>
    </li>
    <li
      className="bg-gray-900 w-32 text-center  my-5 cursor-pointer rounded"
      style={Object {}}
    >
      <button
        className="w-full py-2 focus:outline-none text-white"
      >
        Take a quiz
      </button>
    </li>
    <li
      className="bg-gray-900 w-32 text-center  my-5 cursor-pointer rounded"
      style={Object {}}
    >
      <button
        className="w-full py-2 focus:outline-none text-white"
      >
        About us
      </button>
    </li>
    <h1
      className="absolute bottom-0 mb-10 "
    >
      All Rights Reserved.
    </h1>
  </ul>
  <button
    className="absolute mt-5 mr-2	focus:outline-none top-0 right-0 w-12 h-12 bottom-0 "
    onClick={[Function]}
  >
    <svg
      fill="white"
      height="35"
      stroke="red"
      viewBox="0 0 23 23"
      width="35"
    >
      <path
        d="M 2 2.5 L 20 2.5"
        fill="transparent"
        stroke="hsl(0, 0%, 18%)"
        strokeLinecap="round"
        strokeWidth="3"
        style={Object {}}
      />
      <path
        d="M 5 9.423 L 15 9.423"
        fill="transparent"
        opacity={1}
        stroke="hsl(0, 0%, 18%)"
        strokeLinecap="round"
        strokeWidth="3"
        style={Object {}}
      />
      <path
        d="M 2 16.346 L 20 16.346"
        fill="transparent"
        stroke="hsl(0, 0%, 18%)"
        strokeLinecap="round"
        strokeWidth="3"
        style={Object {}}
      />
    </svg>
  </button>
</nav>
`;
